tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -83,5 +85,76 @@ private string TranslateError (string culture, string errorCode)
 			PropertyInfo propertyInfo = typeof (MSBStrings).GetProperty (errorCode);
 			return (string) propertyInfo.GetValue (null, null);
 		}
+
+		private List<string> commonIgnoreList { get; set; } = null;
+
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void AllErrorTranslation (string culture)
+		{
+			string errorCode = string.Empty;
+			string errorList = string.Empty;
+			bool isFirstRun = false;
+			List<string> cultureIgnoreList = null;
+			List<string> commonValidEntries = new List<string> ();
+			List<string> cultureValidEntries = new List<string> ();
+
+			string fullCommonPath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/common-Translations.ignore";
+			string fullCulturePath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/{culture}-Translations.ignore";
+			string shortCommonPath = "xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/common-Translations.ignore";
+			string shortCulturePath = $"xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/{culture}-Translations.ignore";
+			bool isCommonFileExistant = File.Exists (fullCommonPath);
+			bool isCultureFileExistant = File.Exists (fullCulturePath);
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+
+			if (commonIgnoreList == null && isCommonFileExistant){
+				commonIgnoreList = new List<string> (ReadFile (fullCommonPath));
+				isFirstRun = true;
+			}
```

Comment:
If:

1. ReadFile is changed to handle non-existent files
2. ReadFile is changed to return `IList<string>`
3. The `cultureIgnoreList` is changed to be an `List<string>`

This can be simplified to:

```suggestion
			if (commonIgnoreList == null)
				commonIgnoreList = ReadFile (fullCommonPath);
```

Reply:
@rolfbjarne I am running into casting issues when trying to follow the above. Regarding Step 3, cultureIgnoreList is already a List<string>. Was this possibly a typo? Thanks!

Reply:
So I was able to return IList<string> from ReadFile by appending .ToList() to the return statement and changing Array.Empty<string> () to new List<string> (). 
I also casted the ReadFile calls to make them "ReadFile (fullCommonPath) as List<string>"
Is there a better way to do that?